sdk/rust/src/dstack_client.rs

```diff
@@ -0,0 +1,270 @@
+        self.signature_chain.iter().map(hex::decode).collect()
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct GetQuoteResponse {
+    pub quote: String,
+    pub event_log: String,
+}
+
+impl GetQuoteResponse {
+    pub fn decode_quote(&self) -> Result<Vec<u8>, FromHexError> {
+        hex::decode(&self.quote)
+    }
+
+    pub fn decode_event_log(&self) -> Result<Vec<EventLog>, serde_json::Error> {
+        serde_json::from_str(&self.event_log)
+    }
+
+    pub fn replay_rtmrs(&self) -> Result<HashMap<u8, String>, Box<dyn Error>> {
+        let parsed_event_log: Vec<EventLog> = self.decode_event_log()?;
+        let mut rtmrs = HashMap::new();
+        for idx in 0..4 {
+            let mut history = vec![];
+            for event in &parsed_event_log {
+                if event.imr == idx {
+                    history.push(event.digest.clone());
+                }
+            }
+            rtmrs.insert(idx as u8, replay_rtmr(history)?);
+        }
+        Ok(rtmrs)
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct InfoResponse {
+    app_id: String,
+    instance_id: String,
+    app_cert: String,
+    tcb_info: TcbInfo,
+    app_name: String,
+    public_logs: bool,
+    public_sysinfo: bool,
+    device_id: String,
+    mr_aggregated: String,
+    mr_image: String,
+    mr_key_provider: String,
+    key_provider_info: String,
+    compose_hash: String,
+}
+
+impl InfoResponse {
+    pub fn validated_from_value(mut obj: Value) -> Result<Self, serde_json::Error> {
+        if let Some(tcb_info_str) = obj.get("tcb_info").and_then(Value::as_str) {
+            let parsed_tcb_info: TcbInfo = from_str(tcb_info_str)?;
+            obj["tcb_info"] = serde_json::to_value(parsed_tcb_info)?;
+        }
+        serde_json::from_value(obj)
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct TcbInfo {
+    mrtd: String,
+    rootfs_hash: String,
+    rtmr0: String,
+    rtmr1: String,
+    rtmr2: String,
+    rtmr3: String,
+    event_log: Vec<EventLog>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct GetTlsKeyResponse {
+    pub key: String,
+    pub certificate_chain: Vec<String>,
+}
+
+pub trait BaseClient {}
+
+pub struct DstackClient {
+    base_url: String,
+    endpoint: String,
+    client: ClientKind,
+}
+
+impl BaseClient for DstackClient {}
+
+impl DstackClient {
+    pub fn new(endpoint: Option<&str>) -> Self {
+        let endpoint = get_endpoint(endpoint);
+        let (base_url, client) = match endpoint {
+            ref e if e.starts_with("http://") || e.starts_with("https://") => {
+                (e.to_string(), ClientKind::Http)
+            }
+            _ => ("http://localhost".to_string(), ClientKind::Unix),
+        };
+
+        DstackClient {
+            base_url,
+            endpoint,
+            client,
+        }
+    }
+
+    async fn send_rpc_request(&self, path: &str, payload: &Value) -> Result<Value, Box<dyn Error>> {
```

Comment:
Is it possible to use generic instead of dynamic typing?
Like:
```rust
async fn send_rpc_request<S: Serialize, D: DeserializeOwned>(&self, path: &str, payload: &S) -> anyhow::Result<D>
```